src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackTool.targets

```diff
@@ -74,11 +74,6 @@ NOTE: This file is imported from the following contexts, so be aware when writin
     <_InnerToolsPublishAot Condition="$(_HasRIDSpecificTools) and '$(PublishAot)' == 'true'">true</_InnerToolsPublishAot>
     <PublishAot Condition="!$(_IsRidSpecific) and '$(MSBuildIsRestoring)' != 'true'">false</PublishAot>
 
-    <!-- we want to ensure that we don't publish any AppHosts for the 'outer', RID-agnostic builds -->
-    <UseAppHost Condition="!$(_IsRidSpecific)">false</UseAppHost>
-    <!-- we want to ensure that we _do_ publish any AppHosts for the 'inner', RID-specific builds -->
-    <UseAppHost Condition="$(_IsRidSpecific)">true</UseAppHost>
-
```

Comment:
This is the key change - however once I removed this explicit setting, the `ToolCommandRunner` discovery logic began to assign `dotnet` runner and the `ToolEntryPoint` began to use the managed dll for RID-specific tool packages in the AOT scenario. That's where the rest of the changes to unify on `_ToolUsesPlatformSpecificExecutable` came from.